app/controllers/placements/providers/placements_controller.rb

```diff
@@ -57,6 +60,15 @@ def location_coordinates
     end
   end
 
+  def calculate_travel_time
+    return if search_location.blank?
+
+    Placements::TravelTime.call(
+      origin_address: search_location,
+      destinations: @placements.map(&:school),
```

Comment:
Using a `map` is an expensive process.
It should be more efficient to use:
```suggestion
      destinations: Placements::School.where(id: @placements.select(:school_id)),
```
`map` will also create duplicates of the school record, you would need to do `map(&:school).uniq` to prevent the duplicates. Whereas `where` will only return the uniq matching records.

Reply:
ACTUALLY... I've just tested this myself, and this is a faff if you do it that way. Because `@placements.map(&:school)` passes in the decorated schools associated with the already decorated placements, which is why this works, because the travel times are attributes on the decorated school, not the school model itself. 
At the very least add `.uniq` to the end of the `.map(...)` to remove any duplicated schools.